service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ComplyOnBehalfController.java

```diff
@@ -45,15 +47,23 @@ public ComplyOnBehalfController(ObjectMapper mapper,
         this.othersService = othersService;
     }
 
+    //TODO: filter responses with different userName in aboutToStart. Code below makes the assumption that only
+    // the same responder will be able edit a response. Currently any solicitor can amend a response but the
+    // name of the responder does not change.
     @PostMapping("/about-to-start")
     public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackrequest) {
         CaseDetails caseDetails = callbackrequest.getCaseDetails();
         CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
 
+        List<Element<Direction>> directionsToComplyWith = directionHelperService.getDirectionsToComplyWith(caseData);
+
         Map<DirectionAssignee, List<Element<Direction>>> sortedDirections =
-            directionHelperService.sortDirectionsByAssignee(caseData.getStandardDirectionOrder().getDirections());
+            directionHelperService.sortDirectionsByAssignee(directionsToComplyWith);
+
+        directionHelperService.addEmptyDirectionsForAssigneeNotInMap(sortedDirections);
 
-        directionHelperService.addDirectionsToCaseDetails(caseDetails, sortedDirections);
+        directionHelperService.addDirectionsToCaseDetails(
+            caseDetails, sortedDirections, ComplyOnBehalfEvent.valueOf(callbackrequest.getEventId()));
```

Comment:
I think it should be actually embedded in the service - logic like that is too much for controllr imho, it needs to know empty directions should be added to the map. 

On the other hand it is so tightly coupled with ccd I wouldn't mind having it here massively. What do you think?

Reply:
I don't mind it being here. I guess I good refactor for the future could be to add it to the sortDirectionsByAssignee method? What are your thoughts? Again this is used in more than one place

Reply:
Yeah could be embedded maybe in the sortDirections, we can leave it and refactor later.